src/classes/grade.js

```diff
@@ -1,250 +1,212 @@
-const utils = require('../utils')
-
-const tosdr = require('../../data/generated/tosdr')
-const tosdrMessages = require('../../data/tosdr-messages')
-const majorTrackingNetworks = require('../../data/major-tracking-networks')
+const UNKNOWN_PRIVACY_SCORE = 2
+
+/**
+ * Range map data structures:
+ *
+ * Maps a numeric input to an arbitrary output based on provided ranges
+ *
+ * `steps` defines the range of inputs for each output,
+ * `max` defines what happens if the input is above the given ranges
+ * `zero` is a special case for when the input is 0 or falsy
+ *
+ * For example:
+ *
+ * zero: 'foo',
+ * max: 'qux',
+ * steps: [
+ *     [1, 'bar'],
+ *     [2, 'baz']
+ * ]
+ *
+ * means:
+ *
+ * input <= 0       maps to 'foo'
+ * 0 < input < 1    maps to 'bar'
+ * 1 <= input < 2   maps to 'baz'
+ * input >= 2       maps to 'C'
+ */
+
+const TRACKER_RANGE_MAP = {
+    zero: 0,
+    max: 10,
+    steps: [
+        [0.1, 1],
+        [1, 2],
+        [5, 3],
+        [10, 4],
+        [15, 5],
+        [20, 6],
+        [30, 7],
+        [45, 8],
+        [66, 9]
+    ]
+}
 
-const tosdrRegexList = Object.keys(tosdr).map(x => new RegExp(`(^)${utils.getDomain(x)}`)) // only match domains, and from the start of the URL
-const tosdrClassMap = {'A': -1, 'B': 0, 'C': 0, 'D': 1, 'E': 2} // map tosdr class rankings to increase/decrease in grade
-const siteScores = ['A', 'B', 'C', 'D']
+const GRADE_RANGE_MAP = {
+    zero: 'A',
+    max: 'D-',
+    steps: [
+        [2, 'A'],
+        [4, 'B+'],
+        [10, 'B'],
+        [14, 'C+'],
+        [20, 'C'],
+        [30, 'D']
+    ]
+}
```

Comment:
@russellholt as we discussed yesterday, I created a data structure for the score and grade assignments. It's still O(n) and easier to tweak, but may be less obvious than the if/else structure. I've tried to leave a good comment above.